src/platform/tls_openssl.c

```diff
@@ -1844,14 +1860,14 @@ QuicHpKeyCreate(
         Aead = EVP_aes_256_ecb();
         break;
     case QUIC_AEAD_CHACHA20_POLY1305:
-        Aead = EVP_chacha20_poly1305();
+        Aead = EVP_chacha20();
```

Comment:
So we don't need the poly1305 for header protection?

Reply:
Nope. According to the [QUIC-TLS spec](https://tools.ietf.org/html/draft-ietf-quic-tls-31#section-5.4.4) only the raw ChaCha20 function is used for header protection.